### ISSUE TITLE
"Extension host terminated unexpectedly." error caused by Flutter extension

### ISSUE THREAD

**Issue by MrWillCom** [Dart-Code/Dart-Code#3105]
Issue Type: <b>Bug</b>

To reproduce this problem, you need to install Flutter(`dart-code.flutter`) extension and open an existing Flutter project with Visual Studio Code.

Wait some seconds, you will see a popup notification in the bottom right corner said "Extension host terminated unexpectedly." with options "Open Developer Tools" and "Restart Extension Host":

![a screenshot of the popup error notification](https://user-images.githubusercontent.com/47271684/105952146-20734a00-60ac-11eb-8525-f081411ba1ef.png)

By the way, this error also exists on Visual Studio Code - Insiders.

Extension version: 3.18.1
VS Code version: Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:34:46.910Z)
OS version: Windows_NT x64 10.0.21296

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz (4 x 2304)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (5.32GB free)|
|Process Argv|--crash-reporter-id 26bd872c-b1c3-4c78-af26-9906b4c1ce99|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
openlogontheside:30221877
python383cf:30185419
pythonvspyt700:30244268
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30211401
wsl2promptcf:30224613
vstry244cf:30250196
pythonvsdeb440:30248342
unusedprompt:30224610
folderexplorercf:30224615
openfilemenucf:30224648
pythonvsded773:30248341
pythonvspyt600:30247031

```

</details>

**Comment by DanTup** [comment]
I'm not able to reproduce this - could you click the "Open Developer Tools" button and see if there's anything interesting (errors/warnings) in the Console tab?

**Comment by MrWillCom** [comment]
```
workbench.desktop.main.js:sourcemap:5009 Extension Host
workbench.desktop.main.js:sourcemap:5009 events.js:200
        throw er; // Unhandled 'error' event
        ^

  [Error: read ECONNRESET
  	at TLSWrap.onStreamRead (internal/stream_base_commons.js:200:27)
  Emitted 'error' event on ClientRequest instance at:
  	at TLSSocket.socketErrorListener (_http_client.js:406:9)
  	at TLSSocket.emit (events.js:223:5)
  	at emitErrorNT (internal/streams/destroy.js:92:8)
  	at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
  	at processTicksAndRejections (internal/process/task_queues.js:81:21)] {
    errno: 'ECONNRESET',
    code: 'ECONNRESET',
    syscall: 'read',
    vslsStack: [ CallSite {} ]
  }
workbench.desktop.main.js:sourcemap:5428 Extension host terminated unexpectedly. Code:  1  Signal:  null
  _onExtensionHostCrashed @ workbench.desktop.main.js:sourcemap:5428
workbench.desktop.main.js:sourcemap:1779   ERR Extension host terminated unexpectedly. The following extensions were running: ... Dart-Code.flutter, ...
workbench.desktop.main.js:sourcemap:3183 Extension host terminated unexpectedly.
  onDidChangeNotification @ workbench.desktop.main.js:sourcemap:3183
```

Even it said `Dart-Code.flutter` is running, in fact, it's not running. I tried to start debugging, but Visual Studio Code tells me that it cannot find debug adapter for type 'dart':

![a screenshot of the error dialog popupped when I tried to debug](https://user-images.githubusercontent.com/47271684/105988766-1c125580-60db-11eb-9ebf-c007ecc9e1c3.png)

Also, I tried to run Flutter doctor in Visual Studio Code command palette:

![a screenshot of running Flutter doctor in Visual Studio Code command palette](https://user-images.githubusercontent.com/47271684/105989026-7e6b5600-60db-11eb-9b16-87ed3ae87139.png)

But failed:

![a screenshot of the error message when running Flutter doctor in Visual Studio Code command palette](https://user-images.githubusercontent.com/47271684/105989231-c9856900-60db-11eb-8769-0ce1a96c4c2f.png)

**Comment by DanTup** [comment]
The later errors are likely just because of the initial error. If the extension host crashes, no extensions are available, so anything that tries to interact with them will fail.

If you open another type of project that has extensions (for example TypeScript), does the extension host also crash? I found a few reports of this error that were caused by general VS Code errors (such as antivirus deleting some of its extension host code or other [corruption of VS Code files](https://github.com/Microsoft/vscode/issues/65304#issuecomment-448265950)) - they would affect all extensions equally so it's worth checking if it may be something like that.

**Comment by DanTup** [comment]
A few people in [this SO post](https://stackoverflow.com/questions/46457994/receiving-error-message-extension-host-terminated-unexpectedly-in-visual-stud) mentioned Live Share was causing this for them. Do you have that installed? It may be worth trying with all extensions disabled (or using the `Help: Start Extension Bisect` command to do a binary search of them).

**Comment by MrWillCom** [comment]
I've checked my extensions one by one. Each time, I disable an extension, restart my VS Code and check if extension host terminated unexpectedly. Extension host terminated every time I tried except the time I disabled Flutter extension. So I think I'm sure it's caused by Flutter.

Also, this problem does happen when Live Share is disabled:

![a screenshot of the error message when Live Share is disabled](https://user-images.githubusercontent.com/47271684/105991028-431e5680-60de-11eb-9aed-fd7617197453.png)

**Comment by MrWillCom** [comment]
Versions:

|  | Version |
| --- | --- |
| Flutter SDK | `1.22.5` |
| Dart SDK | `2.10.4` |
| Flutter extension | `v3.18.1` |
| Dart extension| `v3.18.1` |

**Comment by DanTup** [comment]
Can you try:

- uninstall the Flutter extension
- uninstall the Dart extension
- Close VS Code
- Delete any folders starting with `dart-code` from `%USERPROFILE%\.vscode\extensions`
- Re-open VS Code and re-install the Flutter extension (which will also install Dart)
- See if the issue still occurs

This will eliminate any of the extension code being corrupt, which has [happened in the past](https://github.com/microsoft/vscode/issues/73134).

If it still occurs then, try enabling the [Dart-Code general extension log](https://dartcode.org/docs/logging/#extension) which might help understand whether any of the code is bring run at all (and if so, how far it's getting).

Thanks!

**Comment by MrWillCom** [comment]
I followed your instructions and it works!

Thanks for helping me fix this problem! I can use Flutter & Dart extensions in VS Code again now!

Thanks! 😃

**Comment by DanTup** [comment]
Great, glad to hear! It sounds like the extension was somehow corrupt then. I can't explain why, it's not something the extension is involved in (VS code downloads and extracts the extension code and the extension never modifies its own code), but at least you got to the bottom of it! :-)

**Comment by tom-010** [comment]
I get the same error every time, I edit a cookiecutter-project. Maybe the strange directory name causes it - but I did not have a chance to triage the bug more closely.

The top level structure of the cookie-cutter-project looks like:

```bash
cookiecutter.json
{{cookiecutter.project_slug}}
demo
env 
README.md
requirements.txt
scripts
test
```

The `{{cookiecutter.project_slug}}` is a directory containing the blueprint of a flutter app. 

I've noticed that vscode struggles with the ' {{..}}` in other instances as well. 

To reproduce:

1. Open a cookiecutter-project where `{{cookiecutter.project_slug}}` contains a flutter project
2. Don't close this window
3. Open a flutter-project  in another window
4. Get the error "Extension Hosts terminated unexpectedly" as described by the OP.

Temporary fix: Kill **all** VSCode instances, e.g.

```bash
kill $(pgrep -f vscode)
```

If you like, I give it a closer look to find the root of the problem.
